src/main/java/de/tum/in/www1/exerciseapp/web/rest/ExerciseResource.java

```diff
@@ -77,7 +73,7 @@ public ExerciseResource(ExerciseRepository exerciseRepository, ExerciseService e
      * @throws URISyntaxException if the Location URI syntax is incorrect
      */
     @PostMapping("/exercises")
-    @PreAuthorize("hasAnyRole('ADMIN', 'TA')")
+    @PreAuthorize("hasAnyRole('TA', 'INSTRUCTOR', 'ADMIN')")
     @Timed
     //TODO: test if it still works with abstract entity in body
     public ResponseEntity<Exercise> createExercise(@RequestBody Exercise exercise) throws URISyntaxException {
```

Comment:
This should only be possible for TAs and instrcutors of this particular course (and admins)

Reply:
Create exercise feature is currently using POST /api/programming-exercises, where I have inserted the check. I wasn't able to find the usage of this method anywhere and, therefore, there is no check.

Reply:
Done.